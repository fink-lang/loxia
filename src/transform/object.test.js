import {fink2js} from '../testing';


test('object', ()=> {
  expect(
    fink2js(`
      obj1 = {}
      obj2 = {a}
      obj3 = {a, delete, true, false}
      obj4 = {a, b, ...c}
      obj5 = {a: 1, b: 123, \`c-d-e\`: cde }
      obj6 = {a=123, b: c=456}
      obj7 = {a: 123, b: 123 && 123 && 1345, c: fn a, b: 134}
      obj8 = {
        a: 123,
        b: 123
          && 123
          && 1345,
        c: fn a, b: 134,
        d:
          n = a+3
          foo(n)
      }

      {a, b, c}
    `)
  ).toMatchSnapshot();
});


test('destructuring object', ()=> {
  expect(
    fink2js(`
      {a, b, c} = ni
      {e, ...f} = {e: 0, foo: 12, bar: 34}
      {x=1, ni: {y, z}} = {ni: {y: 1, z: 2}}
      {\`n i\`: ni} = foo
    `)
  ).toMatchSnapshot();
});
