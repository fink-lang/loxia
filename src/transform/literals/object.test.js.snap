// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculated props 1`] = `
"const foo = {
  \\"foo\\": foo,
  [Symbol(\`foo\`)]: 123,
  [\`spam\`]: 456,
  \\"ni\\": ni
};
Object.assign(module.exports, {
  foo
});"
`;

exports[`destructuring object 1`] = `
"const {
  \\"a\\": a,
  \\"b\\": b,
  \\"c\\": c
} = ni;
const {
  \\"e\\": e,
  ...f
} = {
  \\"e\\": 0,
  \\"foo\\": 12,
  \\"bar\\": 34
};
const {
  \\"x\\": x = 1,
  \\"ni\\": {
    \\"y\\": y,
    \\"z\\": z
  }
} = {
  \\"ni\\": {
    \\"y\\": 1,
    \\"z\\": 2
  }
};
const {
  [\`n i\`]: ni
} = foo;
Object.assign(module.exports, {});"
`;

exports[`object 1`] = `
"const obj1 = {};
const obj2 = {
  \\"a\\": a
};
const obj3 = {
  \\"a\\": a,
  \\"delete\\": Ë†delete,
  \\"true\\": true,
  \\"false\\": false
};
const obj4 = {
  \\"a\\": a,
  \\"b\\": b,
  ...c
};
const obj5 = {
  \\"a\\": 1,
  \\"b\\": 123,
  [\`c-d-e\`]: cde
};
const obj6 = {
  \\"a\\": a = 123,
  \\"b\\": c = 456
};
const obj7 = {
  \\"a\\": 123,
  \\"b\\": 123 && 123 && 1345,
  \\"c\\": (a, b) => 134
};
const obj8 = {
  \\"a\\": 123,
  \\"b\\": 123 && 123 && 1345,
  \\"c\\": (a, b) => 134,
  \\"d\\": (() => {
    const n = a + 3;
    return foo(n);
  })()
};
({
  \\"a\\": a,
  \\"b\\": b,
  \\"c\\": c
});
Object.assign(module.exports, {
  obj1,
  obj2,
  obj3,
  obj4,
  obj5,
  obj6,
  obj7,
  obj8
});"
`;
